models/DataObject/ClassDefinition/Data/Video.php

```diff
@@ -676,4 +676,42 @@ public function unmarshal($value, $object = null, $params = [])
             return $video;
         }
     }
+
+    /**
+     * @param DataObject\Data\Video|null $oldValue
+     * @param DataObject\Data\Video|null $newValue
+     *
+     * @return bool
+     */
+    public function isEqual($oldValue, $newValue)
+    {
+        if ($oldValue === null && $newValue === null) {
+            return true;
+        }
+
+        if (!$oldValue instanceof DataObject\Data\Video
+            || !$newValue instanceof DataObject\Data\Video) {
+            return false;
+        }
+
+        $oldValue = [
+            'type' => $oldValue->getType(),
+            'data' => $oldValue->getData()
+        ];
+
+        if ($oldValue['data'] instanceof Asset\Video) {
+            $oldValue['data'] = $oldValue['data']->getId();
+        }
+
+        $newValue = [
+            'type' => $newValue->getType(),
+            'data' => $newValue->getData()
+        ];
+
+        if ($newValue['data'] instanceof Asset\Video) {
+            $newValue['data'] = $newValue['data']->getId();
+        }
+
+        return \serialize($oldValue) === \serialize($newValue);
```

Comment:
serialize is not the fastet option, I'd prefer a loop actually comparing values.

Reply:
changed it to loop...thnx!